packages/firebase_auth/firebase_auth_platform_interface/lib/src/platform_interface/platform_interface_multi_factor.dart

```diff
@@ -0,0 +1,163 @@
+    throw UnimplementedError('unenroll() is not implemented');
+  }
+
+  /// Returns a list of the [MultiFactorInfo] already associated with this user.
+  Future<List<MultiFactorInfo>> getEnrolledFactors() {
+    throw UnimplementedError('getEnrolledFactors() is not implemented');
+  }
+}
+
+/// Identifies the current session to enroll a second factor or to complete sign in when previously enrolled.
+///
+/// It contains additional context on the existing user, notably the confirmation that the user passed the first factor challenge.
+class MultiFactorSession {
+  MultiFactorSession(this.id);
+
+  final String id;
+}
+
+/// Represents an assertion that the Firebase Authentication server
+/// can use to authenticate a user as part of a multi-factor flow.
+class MultiFactorAssertionPlatform {}
+
+/// {@macro .platformInterfaceMultiFactorResolverPlatform}
+/// Utility class that contains methods to resolve second factor
+/// requirements on users that have opted into two-factor authentication.
+/// {@endtemplate}
+class MultiFactorResolverPlatform {
+  /// {@macro .platformInterfaceMultiFactorResolverPlatform}
+  const MultiFactorResolverPlatform(
+    this.hints,
+    this.session,
+  );
+
+  /// List of [MultiFactorInfo] which represents the available
+  /// second factors that can be used to complete the sign-in for the current session.
+  final List<MultiFactorInfo> hints;
+
+  /// A MultiFactorSession, an opaque session identifier for the current sign-in flow.
+  final MultiFactorSession session;
+
+  /// Completes sign in with a second factor using an MultiFactorAssertion which
+  /// confirms that the user has successfully completed the second factor challenge.
+  Future<UserCredentialPlatform> resolveSignIn(
+    MultiFactorAssertionPlatform assertion,
+  ) {
+    throw UnimplementedError('resolveSignIn() is not implemented');
+  }
+}
+
+/// Represents a single second factor means for the user.
+///
+/// See direct subclasses for type-specific information.
+class MultiFactorInfo {
+  const MultiFactorInfo({
+    required this.factorId,
+    required this.enrollmentTimestamp,
+    required this.displayName,
+    required this.uid,
+  });
+
+  /// User-given display name for this second factor.
+  final String? displayName;
+
+  /// The enrollment timestamp for this second factor in seconds since epoch (UTC midnight on January 1, 1970).
+  final double enrollmentTimestamp;
+
+  /// The factor id of this second factor.
+  final String factorId;
+
+  /// The unique identifier for this second factor.
+  final String uid;
+}
+
+/// Represents the information for a phone second factor.
+class PhoneMultiFactorInfo extends MultiFactorInfo {
+  const PhoneMultiFactorInfo({
+    required String? displayName,
+    required double enrollmentTimestamp,
+    required String factorId,
+    required String uid,
+    required this.phoneNumber,
+  }) : super(
+          displayName: displayName,
```

Comment:
Are we using Dart 2.17 already? If so, these should be using the new super syntax

Reply:
nvm we aren't